syntax = "proto3";

////////////////////////////////////////////
///////////////BASIC RESPONSES//////////////
////////////////////////////////////////////

message UserData {
    string username = 1;
    string password = 2;
}

message Success {
	bool success = 1;
}

message Empty {

}
////////////////////////////////////////////
////////////////////////////////////////////
////////////////////////////////////////////

////////////////////////////////////////////
////////////////BASIC REQUESTS//////////////
////////////////////////////////////////////

message UserDataRequest {
	string username = 1;
	string password = 2;
}

message UserRequest {
	string username = 1;
}

message UpdateUserRequest {
	string username = 1;
	string new_username = 2;
	string new_password = 3;
}
////////////////////////////////////////////
////////////////////////////////////////////
////////////////////////////////////////////

service SignIn {
	rpc CreateUser (UserDataRequest) returns (Success);
	rpc LoginUser (UserDataRequest) returns (Success);
	rpc LogoutUser (Empty) returns (Success);
	rpc GetUserByName (UserRequest) returns (UserData);
	rpc UpdateUser (UpdateUserRequest) returns (Success);
	rpc DeleteUser (UserRequest) returns (Success);
}